packages/vx-tooltip/src/enhancers/withTooltip.tsx

```diff
@@ -1,28 +1,40 @@
 import React from 'react';
-import PropTypes from 'prop-types';
 
-export const withTooltipPropTypes = {
-  tooltipOpen: PropTypes.bool,
-  tooltipLeft: PropTypes.number,
-  tooltipTop: PropTypes.number,
-  tooltipData: PropTypes.object,
-  updateTooltip: PropTypes.func,
-  showTooltip: PropTypes.func,
-  hideTooltip: PropTypes.func,
+type Props = {
+  tooltipOpen?: boolean;
+  tooltipLeft?: number;
+  tooltipTop?: number;
+  tooltipData?: Object;
+  updateTooltip?: (args: UpdateTooltipArgs) => void;
+  showTooltip?: (args: ShowTooltipArgs) => void;
+  hideTooltip?: () => void;
+};
+
+type ShowTooltipArgs = {
+  tooltipLeft?: Props['tooltipLeft'];
+  tooltipTop?: Props['tooltipTop'];
+  tooltipData?: Props['tooltipData'];
+};
+
+type UpdateTooltipArgs = {
+  tooltipOpen?: Props['tooltipOpen'];
+  tooltipLeft?: Props['tooltipLeft'];
+  tooltipTop?: Props['tooltipTop'];
+  tooltipData?: Props['tooltipData'];
 };
 
 export default function withTooltip(
```

Comment:
I think the `props` aren't quite right for this. For this to be compatible with any `BaseComponent` there will need to be a `Generic` here for the `BaseComponent`s own props. Something like this:

```jsx
function withTooltip<Props extends object = {}>(
  BaseComponent: (React.Component | React.PureComponent | React.FC)<ProvidedProps & Props>,
  containerProps: WithTooltipContainerProps = ...,
) {
  class WrappedComponent extends React.PureComponent<Props, WithTooltipState> {
     ...
  }
}
```

@milesj does this look right?